pkg/services/panellib/panellib.go

```diff
@@ -0,0 +1,61 @@
+package panellib
+
+import (
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// PanelLib is the service the Panel Library feature.
+type PanelLib struct {
+	Cfg      *setting.Cfg       `inject:""`
+	SQLStore *sqlstore.SQLStore `inject:""`
+	log      log.Logger
+}
+
+func init() {
+	registry.RegisterService(&PanelLib{})
+}
+
+// Init initializes the PanelLib Service.
+func (pl *PanelLib) Init() error {
+	pl.log = log.New("panellib")
+
+	return nil
+}
+
+// IsEnabled returns true if the PanelLib service is enabled for this instance.
+func (pl *PanelLib) IsEnabled() bool {
+	if pl.Cfg == nil {
+		return false
+	}
+
+	return pl.Cfg.IsPanelLibraryEnabled()
+}
+
+// AddMigration defines database migrations.
+// If Panel Library is not enabled does nothing.
+func (pl *PanelLib) AddMigration(mg *migrator.Migrator) {
+	if !pl.IsEnabled() {
+		return
+	}
+
+	panellibV1 := migrator.Table{
+		Name: "panellib",
```

Comment:
panel_lib ? the entity/model name feels like it will be LibraryPanel, or PanelLibraryItem ? 

Hm.. we can also think about this with global variables in mind. maybe the tables are just panel and (variable for future global variables)?  

So 
* panellib  (hard to read with the double lowercase l)
* panel_lib
* library_panel
* library_variable
* library_dashboard_template
* panel

naming is hard :)

Reply:
I think you make a valid point, my first thought was `panel` as well but I dropped it. I think the `library_xyz`naming is better so I'll change. Thank you